src/reducers/entities/teams.js

```diff
@@ -210,16 +210,13 @@ function myMembers(state = {}, action) {
     }
 
     case ChannelTypes.POST_UNREAD_SUCCESS: {
-        const {teamId, deltaMsgs, mentionCount} = action.data;
-        const teamState = state[teamId];
-        if (!teamState) {
-            return state;
-        }
+        const {teamId, deltaMsgs, mentionCount, msgCount} = action.data;
+        const teamState = (typeof state[teamId] === 'undefined' ? {} : state[teamId]);
 
         const newTeamState = {
             ...teamState,
-            msg_count: teamState.msg_count - deltaMsgs,
-            mention_count: teamState.mention_count + mentionCount,
+            msg_count: (typeof teamState.msg_count === 'undefined' ? msgCount : teamState.msg_count - deltaMsgs),
```

Comment:
Perhaps it would actually be cleaner to make two separate actions so that you can identify what fields should exist on each one? It now seems like we have two types of `POST_UNREAD_SUCCESS` actions with different meanings that contain different fields

Reply:
I see it the other way around, I have 1 action with different effects on different sections on the state, not sure how it would be more clear having multiple actions, as you then would have to track all the info.

Another option would be to switch to something closer to `sagas` where each action would only have a few side effects, but we would then have to rethink the whole repo in a way that matches that architecture.

Reply:
Does the action always contain both fields? Since you're null checking it, I was thinking that it contains either one field or the other. If the action always contains both, then I think we're fine to keep using one action, but then we could get rid of the null checking to make it clearer

Reply:
it should contain both, but running the tests gave me a bit of a headache and sometimes there were missing elements, so I checked for them and choose a default value, which might not make sense (I'm not happy with default for msg_count as I'm not sure it won't casuse trouble, I'm fine with mentions).

Reply:
Ah, I see. I'd lean towards making the tests a bit worse to make the rest of the code better, but it's your call then.